modules/channel-web/src/backend/socket.ts

```diff
@@ -38,15 +38,18 @@ export default async (bp: typeof sdk, db: Database) => {
       const payload = bp.RealTimePayload.forVisitor(visitorId, 'webchat.data', event.payload)
       bp.realtime.sendPayload(payload)
     } else {
-      if (event.payload.typing !== false) {
-        const payload = bp.RealTimePayload.forVisitor(visitorId, 'webchat.typing', { timeInMs: 500, conversationId })
+      if (event.payload.typing === true || event.payload.type === 'typing') {
+        const value = event.payload.type === 'typing' ? event.payload.value : 500
```

Comment:
I think `event.payload.value || 500` would be better. I'm not sure if we necessarly want to restrict this to only typing events, but if we do in that case the condition should be `(event.payload.type === 'typing' ? event.payload.value : undefined) || 500` otherwise a typing event with an undefined value would not work